tests/functional/aws-node-sdk/test/multipleBackend/put.js

```diff
@@ -0,0 +1,181 @@
+                };
+                s3.putObject(params, err => {
+                    assert.equal(err, null, 'Expected success, ' +
+                        `got error ${JSON.stringify(err)}`);
+                    done();
+                });
+            });
+
+            it('should put an object to file', done => {
+                const params = { Bucket: bucket, Key: key,
+                    Body: body,
+                    Metadata: { 'scal-location-constraint': 'file' },
+                };
+                s3.putObject(params, err => {
+                    assert.equal(err, null, 'Expected success, ' +
+                        `got error ${JSON.stringify(err)}`);
+                    done();
+                });
+            });
+        });
+    });
+});
+
+describe('MultipleBackend put object based on bucket location', () => {
+    withV4(sigCfg => {
+        const params = { Bucket: bucket, Key: key, Body: body };
+
+        afterEach(() => {
+            process.stdout.write('Emptying bucket\n');
+            return bucketUtil.empty(bucket)
+            .then(() => {
+                process.stdout.write('Deleting bucket\n');
+                return bucketUtil.deleteOne(bucket);
+            })
+            .catch(err => {
+                process.stdout.write(`Error in afterEach: ${err}\n`);
+                throw err;
+            });
+        });
+
+        it('should put an object mem with no location header', done => {
+            bucketUtil = new BucketUtility('default', sigCfg);
+            s3 = bucketUtil.s3;
+            process.stdout.write('Creating bucket\n');
+            return s3.createBucket({ Bucket: bucket,
+                CreateBucketConfiguration: {
+                    LocationConstraint: 'mem',
+                },
+            }, err => {
+                if (err) {
+                    process.stdout.write(`Error creating bucket: ${err}\n`);
+                    return err;
+                }
+                process.stdout.write('Putting object\n');
+                return s3.putObject(params, err => {
+                    assert.equal(err, null, 'Expected success, ' +
+                        `got error ${JSON.stringify(err)}`);
+                    s3.getObject({ Bucket: bucket, Key: key }, err => {
+                        assert.strictEqual(err, null, 'Expected success, ' +
+                        `got error ${JSON.stringify(err)}`);
+                        done();
+                    });
+                });
+            });
+        });
+    });
+});
+
+describe('MultipleBackend put based on request endpoint', () => {
+    withV4(sigCfg => {
+        after(() => {
+            process.stdout.write('Deleting bucket\n');
+            return bucketUtil.deleteOne(bucket)
+            .catch(err => {
+                process.stdout.write(`Error in after: ${err}\n`);
+                throw err;
+            });
+        });
+
+        it('should create bucket in corresponding backend', done => {
```

Comment:
shouldn't the test really be about putting an object once the bucket is created without a locationConstraint?  I think you need to take the next step and put the object.